server/src/main/java/io/mantisrx/master/api/akka/route/v1/JobsRoute.java

```diff
@@ -228,9 +242,17 @@ private Route getJobsRoute(Optional<String> clusterName) {
                                     clusterName.map(s -> Optional.of("^" + s + "$")).orElse(matching),
                                     true);
 
+                              JobClusterManagerProto.ListJobsResponse cachedResponse = listJobsCache.getIfPresent(listJobsRequest);
+
                               return completeAsync(
-                                    jobRouteHandler.listJobs(listJobsRequest),
-                                    resp -> completeOK(
+                                    cachedResponse == null ? jobRouteHandler.listJobs(listJobsRequest) : CompletableFuture.completedFuture(cachedResponse),
+                                    resp -> {
+
+                                        if (config.getApiCacheEnabled() && cachedResponse == null) {
+                                            listJobsCache.put(listJobsRequest, resp);
+                                        }
```

Comment:
I chose this approach instead of the `Cache#get(key, supplier)` approach for two reasons:

1. I wasn't sure if the `CompletionStage<>` from `jobRouteHandler#listJobs` would become a `CompletedFuture<>` or equivalent afterwards or become unusable.

2. The `Cache#get` method throws a `java.util.concurrent.ExecutionException` and I was of the opinion that this was less ugly than dealing with the exception.